packages/core/http/core-http-router-server-internal/src/validator.ts

```diff
@@ -18,6 +19,10 @@ import type {
 } from '@kbn/core-http-server';
 import { RouteValidationError } from '@kbn/core-http-server';
 
+export function isZod<T>(rule: RouteValidationSpec<T>): rule is z.ZodType<T> {
+  return Boolean('_def' in rule && 'typeName' in (rule as { _def: {} })._def);
```

Comment:
Would this be simpler:
```
zodShape instanceof z.Schema<any>;
```

Based of https://github.com/colinhacks/zod/issues/170#issuecomment-699694602 (and I tried this while working on adding zod support to the route repository package)

Reply:
Using this sounds good to me! 👍🏻 
@jloleysens any objection?

Reply:
Done and refactored to it living `@kbn/zod`